### PR TITLE
add warning for known problematic env vars

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

File: scripts/newinstall.sh [hunk 1]
```diff
@@ -784,6 +784,49 @@ n8l::install_eups() {
...
+	echo -n "${found[@]}"
+}
+
+n8l::problem_vars_check() {
+	local problems=()
+	problems=($(n8l::problem_vars))
+
+	if [[ ${#problems} -gt 0 ]]; then
+		n8l::print_error "$(cat <<-EOF
+			WARNING: the following environment variables are defined that will effect
```
Comment (timj) [inline scripts/newinstall.sh]: "affect"

File: scripts/newinstall.sh [hunk 2]
```diff
@@ -784,6 +784,49 @@ n8l::install_eups() {
...
+			EOF
+		)"
+
+		for v in ${problems[*]}; do
+			n8l::print_error "${v}=\"${!v}\""
+		done
+
+		n8l::print_error "$(cat <<-EOF
+
+			It is recommend that they are undefined before running this script.
```
Comment (timj) [inline scripts/newinstall.sh]: "recommended"

**Comment by jhoblitt** [comment]
I'm not sure why? `rspec` is no more verbose than `unittest` would be with a couple of examples per method with 3-4 assertions and some mocking.

**Comment by jhoblitt** [comment]
Ugh.  This needs a rethink as `LSST_HOME` is defined in the script.

**Pull request merged**